roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set gateway address for multi subnet 
+  set_fact: dfl_nw_gw="{{ item.gateway }}"
+  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') ))) 
+  with_items: "{{ undercloud_cidr }}"
+
+- name: add static route for multi subnet
+  command: "/sbin/ip route replace {{ item.cidr }} via {{ dfl_nw_gw }}"
+  when: (update_routes|default('false')|bool) and (not (ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') )))
+  with_items: "{{ undercloud_cidr }}"
+
+- name: Check if interface has its own config file
+  stat: path=/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg
+  register: found_in_interfaces_d 
+  when: (update_routes|default('false')|bool)
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg"
+  when: (update_routes|default('false')|bool) and found_in_interfaces_d.stat.exists is defined and found_in_interfaces_d.stat.exists
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces"
+  when: (update_routes|default('false')|bool) and ( not found_in_interfaces_d.stat.exists|bool )
```

Comment:
Instead of trying to figure out which file to grep, why not just grep all of them? Grep can take more than one file as input.